lib/semantics/resolve-names.cc

```diff
@@ -2799,15 +2799,32 @@ bool SubprogramVisitor::BeginMpSubprogram(const parser::Name &name) {
   return true;
 }
 
+// A subprogram declared with SUBROUTINE or function
+// If the subprogram is not in interface block, check if its declared as separate 
+// module procedures
 bool SubprogramVisitor::BeginSubprogram(
     const parser::Name &name, Symbol::Flag subpFlag, bool hasModulePrefix) {
+  bool isSeparateModuleProc = false;
+  // Check if Subprogram has Module Prefix and Subprogram is not inside Interface Block 
   if (hasModulePrefix && !inInterfaceBlock()) {
-    Say(name, "'%s' was not declared a separate module procedure"_err_en_US);
-    return false;
+    // Check if the Subprogram is declared as separate module procedures
+    auto *symbol{FindSymbol(name)};
+    if (symbol && symbol->has<SubprogramNameDetails>()) {
```

Comment:
Why do you need to do this check rather than just looking directly in the parent scope?

Reply:
In couple of places before searching in parent scope, this check present. i could not get a scenario where this check is valid/invalid so i just retained it here as well. 
I will remove this check.